src/realizations/catchment/Bmi_Py_Formulation.cpp

```diff
@@ -63,12 +63,38 @@ string Bmi_Py_Formulation::get_output_line_for_timestep(int timestep, std::strin
     // This probably should never happen, but just to be safe ...
     if (output_var_names.empty()) { return ""; }
 
+    // TODO: add codes to handle both the scalar and vector outputs
+    time_t t_delta = timestep;
     // Do the first separately, without the leading comma
-    *output_text_stream << get_var_value_as_double(output_var_names[0]);
+    // The following cases has beeb tested:
+    //*output_text_stream << get_var_vec_as_double(t_delta, output_var_names[0])[0];
+    //*output_text_stream <<"," << get_var_vec_as_double(t_delta, output_var_names[0])[1];
+    //*output_text_stream <<"," << get_var_vec_as_double(t_delta, output_var_names[0])[17646500];
+
+    // Using array to select indices of grid points for output data
+    int array[5] = {0,1,17646300,17646500,17646000};
```

Comment:
Remove test data... lets keep subsetting of output out of this PR.

Reply:
OK. It is a validation that vectorized/gridded version works.